feat: add Azure Blob exporter (#4447)
grafana/alloy: docs/sources/reference/components/otelcol/otelcol.exporter.azureblob.md

---
canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.azureblob/
description: Learn about otelcol.exporter.azureblob
labels:
  stage: experimental

>   products:
>     - oss
> title: otelcol.exporter.azureblob

Reviewer: ```suggestion
  products:
    - oss
  tags:
    - text: Community
      tooltip: This component is developed, maintained, and supported by the Alloy user community.
title: otelcol.exporter.azureblob
```

Add the community supported component badge

---

> # `otelcol.exporter.azureblob`
> 

Reviewer: ```suggestion
# `otelcol.exporter.azureblob`

{{< docs/shared lookup="stability/community.md" source="alloy" version="<ALLOY_VERSION>" >}}

```

Add community supported note

{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}

`otelcol.exporter.azureblob` receives telemetry data from other `otelcol` components and writes it to Azure Blob Storage.

{{< admonition type="note" >}}
`otelcol.exporter.azureblob` is a wrapper over the upstream OpenTelemetry Collector [`azureblob`][] exporter.
Bug reports or feature requests will be redirected to the upstream repository if necessary.

[`azureblob`]: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/{{< param "OTEL_VERSION" >}}/exporter/azureblobexporter
{{< /admonition >}}

You can specify multiple `otelcol.exporter.azureblob` components by giving them different labels.

## Usage

```alloy
otelcol.exporter.azureblob "<LABEL>" {
  blob_uploader {
    url = "https://<ACCOUNT>.blob.core.windows.net"
    container {
      logs    = "otel-logs"
      metrics = "otel-metrics"
      traces  = "otel-traces"
    }
  }
}
```

## Arguments

> This component doesn't support any top-level arguments.

Reviewer: ```suggestion
The `otelcol.exporter.azureblob` component doesn't support any arguments. You can configure this component with blocks.
```

---
canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.azureblob/
description: Learn about otelcol.exporter.azureblob
labels:

>   stage: experimental

Reviewer: ```suggestion
labels:
```

  products:
    - oss
  tags:
    - text: Community
      tooltip: This component is developed, maintained, and supported by the Alloy user community.
title: otelcol.exporter.azureblob
---

# `otelcol.exporter.azureblob`

{{< docs/shared lookup="stability/community.md" source="alloy" version="<ALLOY_VERSION>" >}}

{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}

> `otelcol.exporter.azureblob` receives telemetry data from other `otelcol` components and writes it to Azure Blob Storage.

Reviewer: ```suggestion
`otelcol.exporter.azureblob` receives telemetry data from other `otelcol` components and writes it to Azure Blob Storage.
```

Remove experimental shared link

| `timeout` | `duration` | Timeout for each attempt to send data to Azure.  | `"30s"`  | no       |
| `url`     | `string`   | Azure Storage account URL.                       |          | no       |

The `format` argument must be either `json` or `proto`.
The `url` argument is required unless `auth.type` is `connection_string`.

## Blocks

You can use the following blocks with `otelcol.exporter.azureblob`:

{{< docs/alloy-config >}}

| Block                                  | Description                                                                    | Required |
| -------------------------------------- | ------------------------------------------------------------------------------ | -------- |
| [`auth`][auth]                         | Configures Azure authentication.                                               | yes      |
| [`container`][container]               | Configures the container name for metrics, logs, and traces.                   | no       |
| [`blob_name_format`][blob_name_format] | Configures the blob name format.                                               | no       |

> | [`append_blob`][append_blob]           | Enables append blob mode and separator.                                        | no       |

Reviewer: ```suggestion
| [`append_blob`][append_blob]           | Enables append blob mode and separator.                                        | no       |
| [`blob_name_format`][blob_name_format] | Configures the blob name format.                                               | no       |
| [`container`][container]               | Configures the container name for metrics, logs, and traces.                   | no       |
```

Alpha sort rows

| `metrics` | `string` | Container name for metrics. | `"metrics"` | no       |
| `logs`    | `string` | Container name for logs.    | `"logs"`    | no       |
| `traces`  | `string` | Container name for traces.  | `"traces"`  | no       |

### `blob_name_format`

The `blob_name_format` block configures blob names.

The following arguments are supported:

| Name                          | Type                | Description                                       | Default                              | Required |
| ----------------------------- | ------------------- | ------------------------------------------------- | ------------------------------------ | -------- |
| `metrics_format`              | `string`            | Blob name format for metrics.                     | `"2006/01/02/metrics_15_04_05.json"` | no       |
| `logs_format`                 | `string`            | Blob name format for logs.                        | `"2006/01/02/logs_15_04_05.json"`    | no       |
| `traces_format`               | `string`            | Blob name format for traces.                      | `"2006/01/02/traces_15_04_05.json"`  | no       |

> | `serial_num_enabled`          | `boolean`           | Whether to append a random serial number.         | `true`                               | no       |

Reviewer: ```suggestion
| `serial_num_enabled`          | `bool`              | Whether to append a random serial number.         | `true`                               | no       |
```

Use bool instead of boolean

| `serial_num_range`            | `int`               | Upper limit for the random serial suffix.         | `10000`                              | no       |

> | `serial_num_before_extension` | `boolean`           | Place serial before file extension.               | `false`                              | no       |

Reviewer: ```suggestion
| `serial_num_before_extension` | `bool`              | Place serial before file extension.               | `false`                              | no       |
```

| `timezone`                    | `string`            | Timezone used when formatting blob names.         |                                      | no       |
| `template_enabled`            | `boolean`           | Enable Go template expansion in blob name format. | `false`                              | no       |

> | `time_parser_enabled`         | `boolean`           | Enable time parsing in blob name format.          | `true`                               | no       |

Reviewer: ```suggestion
| `template_enabled`            | `bool`              | Enable Go template expansion in blob name format. | `false`                              | no       |
| `time_parser_enabled`         | `bool`              | Enable time parsing in blob name format.          | `true`                               | no       |
```

| `time_parser_ranges`          | `list(string)`      | Time ranges used by the time parser.              |                                      | no       |
| `params`                      | `map(string)`       | Additional template parameters.                   | `{}`                                 | no       |

### `append_blob`

The `append_blob` block configures append blob behavior.

The following arguments are supported:

| Name        | Type      | Description                            | Default | Required |
| ----------- | --------- | -------------------------------------- | ------- | -------- |

> | `enabled`   | `boolean` | Enable append blob mode.               | `false` | no       |

Reviewer: ```suggestion
| `enabled`   | `bool`    | Enable append blob mode.               | `false` | no       |
```

| `separator` | `string`  | Separator used when appending content. | `"\n"`  | no       |

### `debug_metrics`

{{< docs/shared lookup="reference/components/otelcol-debug-metrics-block.md" source="alloy" version="<ALLOY_VERSION>" >}}

### `retry_on_failure`

The `retry_on_failure` block configures retry backoff for failed requests.

The following arguments are supported:

| Name                   | Type       | Description                              | Default | Required |
| ---------------------- | ---------- | ---------------------------------------- | ------- | -------- |

> | `enabled`              | `boolean`  | Enable retries.                          | `true`  | no       |

Reviewer: ```suggestion
| `enabled`              | `bool`     | Enable retries.                          | `true`  | no       |
```
